Automattic/jetpack
# AI Assistant: Enable inline Heading extension

Fixes #37385

## Proposed changes:

* Adds an override to enable specific extensions without the feature flag, including the heading extension
* Generalizes the BlockHandler to allow extending other blocks later on

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test on JN, Simple and P2
* Keep the blocks variation as "production" and do not enable the feature flag
* Go to the block editor and add a Heading block
* Check that the inline extension is enabled and working as expected
* Check that the other extensions (paragraph and list) are unchanged and working as expected
